### PR TITLE
feat: unicorn now can output ising models

### PR DESCRIPTION
**Pull request by smml1996**

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -203,7 +203,12 @@ pub fn args() -> Command<'static> {
                     .value_name("NUMBER")
                     .validator(is::<usize>),
                 )
-
+                .arg(
+                    Arg::new("ising")
+                    .help("Perform bitblasting of the model")
```
Comment (mstarzinger) [inline src/cli.rs]: The description of the option seems to be outdated.

File: src/unicorn/qubot.rs [hunk 2]
```diff
@@ -377,27 +412,16 @@ impl<'a> Qubot<'a> {
 
         writeln!(out)?;
 
-        // TODO: If Qubits are always referenced by their `name: u64` this can
-        // be simplified by just typedef'ing `Qubit` instead.
-        let mut sorted_linear_coeffs: Vec<(&HashableQubitRef, &i32)> =
-            self.qubo.linear_coefficients.iter().collect();
-        sorted_linear_coeffs.sort_by_key(|(qubit, _)| (*qubit.value.borrow()).name);
-        for (qubit, coeff) in sorted_linear_coeffs {
+        for (qubit, coeff) in self.qubo.linear_coefficients.iter() {
```
Comment (mstarzinger) [inline src/unicorn/qubot.rs]: This undoes my fix that sorts coefficients by the Qubit number. Can we restore that? I think it is still needed.

File: src/unicorn/qubot.rs [hunk 3]
```diff
@@ -377,27 +412,16 @@ impl<'a> Qubot<'a> {
...
             writeln!(out, "{} {}", id, *coeff)?;
         }
 
         writeln!(out)?;
-
-        let mut sorted_quadratic_coeffs: Vec<(&HashableQubitRef, &HashMap<HashableQubitRef, i32>)> =
-            self.qubo.quadratic_coefficients.iter().collect();
-        sorted_quadratic_coeffs.sort_by_key(|(qubit, _)| (*qubit.value.borrow()).name);
-        for (qubit1, edges) in sorted_quadratic_coeffs {
+        for (qubit1, edges) in self.qubo.quadratic_coefficients.iter() {
```
Comment (mstarzinger) [inline src/unicorn/qubot.rs]: Likewise.

File: src/unicorn/qubot.rs [hunk 4]
```diff
@@ -377,27 +412,16 @@ impl<'a> Qubot<'a> {
...
             let id1 = (*qubit1.value.borrow()).name;
 
-            let mut sorted_edges: Vec<(&HashableQubitRef, &i32)> = edges.iter().collect();
-            sorted_edges.sort_by_key(|(qubit, _)| (*qubit.value.borrow()).name);
-            for (qubit2, coeff) in sorted_edges {
+            for (qubit2, coeff) in edges.iter() {
```
Comment (mstarzinger) [inline src/unicorn/qubot.rs]: Likewise.

**Review by mstarzinger — CHANGES_REQUESTED** [review]
Nice! Looks good to me, just some minor comments.

**Review by mstarzinger — APPROVED** [review]
Nice! Looks good to me.

**Comment by smml1996** [comment]
> Nice! Looks good to me.

I just reverted so that I do not erase your changes

**Pull request merged**